addons/survey/static/src/js/survey_form.js

```diff
@@ -146,11 +148,17 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend({
         }
 
         var $matrixBtn = $target.closest('.o_survey_matrix_btn');
+
         if ($target.attr('type') === 'radio') {
             var isQuestionComplete = false;
             if ($matrixBtn.length > 0) {
-                $matrixBtn.closest('tr').find('td').removeClass('o_survey_selected');
-                $matrixBtn.addClass('o_survey_selected');
+                if($matrixBtn.hasClass('o_survey_selected')){
+                    $matrixBtn.removeClass('o_survey_selected');
+                } else {
+                    $matrixBtn.closest('tr').find('td').removeClass('o_survey_selected');
```

Comment:
Can you add a comment on top of this line?
`// uncheck the row of other options since we can only select one`

It took me a minute to understand why we needed that, a comment would not hurt.